engines/trecision/actor.cpp

```diff
@@ -0,0 +1,481 @@
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.000000f, 0.000000f},
+	{-0.647869f, 0.041892f, 0.628372f}, {0.000000f, 0.000000f, 0.000000f},
+	{0.000000f, -0.442684f, -0.328400f}, {-0.294485f, 1.026345f, -1.005401f},
+	{-0.353383f, -1.633775f, -1.214859f}, {-0.312389f, 0.000000f, 0.000000f},
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.020947f, -0.146637f},
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.020946f, -0.146637f},
+	{0.000000f, 0.020946f, -0.083786f}, {0.000000f, 0.020947f, -0.146637f},
+	{0.000000f, 0.020947f, -0.125687f}, {0.000000f, 0.020947f, -0.083786f},
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.020947f, -0.125687f},
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.020946f, -0.125687f},
+	{0.000000f, 0.020946f, -0.146637f}, {0.000000f, 0.000000f, 0.000000f},
+	{0.000000f, 0.000000f, 0.000000f}, {0.000000f, 0.000000f, 0.000000f}};
+
+static const int _vertsCorrList[84] = {
+	289, 290, 293, 294, 295, 296, 297, 298,
+	299, 300, 300, 302, 303, 304, 305, 305,
+	307, 307, 309, 310, 311, 312, 313, 314,
+	315, 316, 317, 318, 319, 320, 321, 322,
+	323, 324, 325, 326, 327, 328, 329, 330,
+	331, 332, 333, 334, 335, 336, 337, 338,
+	339, 340, 341, 349, 350, 352, 353, 354,
+	355, 356, 357, 358, 359, 360, 361, 362,
+	363, 364, 365, 366, 367, 368, 369, 370,
+	371, 372, 373, 374, 375, 376, 377, 378,
+	379, 380, 381, 382};
+
+void Actor::readModel(const char *filename) {
+	Common::SeekableReadStream *ff = _vm->_dataFile.createReadStreamForMember(filename);
+	if (ff == nullptr)
+		error("readModel - Error opening file %s", filename);
+
+	uint32 actionNum = ff->readUint32LE();
+	_vertexNum = ff->readUint32LE();
+
+	_characterArea = new SVertex[_vertexNum * actionNum];
+	for (uint i = 0; i < _vertexNum * actionNum; ++i) {
+		_characterArea[i]._x = ff->readFloatLE();
+		_characterArea[i]._y = ff->readFloatLE();
+		_characterArea[i]._z = ff->readFloatLE();
+		_characterArea[i]._nx = ff->readFloatLE();
+		_characterArea[i]._ny = ff->readFloatLE();
+		_characterArea[i]._nz = ff->readFloatLE();
+	}
+	_vertex = _characterArea;
+	_faceNum = ff->readUint32LE();
+	delete ff;
+
+	ff = _vm->_dataFile.createReadStreamForMember("mat.tex");
+	if (ff == nullptr)
+		error("readModel - Error opening file mat.tex");
+
+	for (uint16 i = 0; i < 256; ++i) {
+		for (uint16 j = 0; j < 91; ++j)
+			_textureMat[i][j] = ff->readUint16LE();
+	}
+
+	_vm->_graphicsMgr->updatePixelFormat((uint16 *)_textureMat, 91 * 256);
+
+	for (uint16 i = 0; i < MAXFACE; ++i) {
+		for (uint16 j = 0; j < 3; ++j) {
+			_textureCoord[i][j][0] = ff->readSint16LE();
+			_textureCoord[i][j][1] = ff->readSint16LE();
+		}
+	}
+
+	_face = new SFace[_faceNum];
+	for (uint i = 0; i < _faceNum; ++i) {
+		_face[i]._a = ff->readUint16LE();
+		_face[i]._b = ff->readUint16LE();
+		_face[i]._c = ff->readUint16LE();
+		_face[i]._mat = ff->readUint16LE();
+	}
+
+	delete ff;
+
+	_curFrame = 0;
+	_curAction = hSTAND;
+
+	// fixup Microprose head correction
+#define P1 306
```

Comment:
I have a slight preference for using `static const int` for this sort of thing as it should compile to the same thing but avoids the potential problems of defines which stick around for the rest of the file

Reply:
+1 as this is not part of the namespace and will potentially break the unity cmake build that I'm using from time to time.

Reply:
I've moved that fix in a separate function and turn those defines into static const as suggested.